gillespy2/solvers/cpp/ode_c_solver.py

```diff
@@ -39,17 +39,33 @@ def run(self=None, model: Model = None, t: int = None, number_of_trajectories: i
             increment: int = None, seed: int = None, debug: bool = False, profile: bool = False, variables={},
             resume=None, live_output: str = None, live_output_options: dict = {}, **kwargs):
 
+        from gillespy2 import log
+
         if self is None:
+            # Post deprecation block
+            # raise SimulationError("ODECSolver must be instantiated to run the simulation")
+            # Pre deprecation block
+            log.warning(
+                """
+                `gillespy2.Model.run(solver=ODECSolver)` is deprecated.
+
+                You should use `gillespy2.Model.run(solver=ODECSolver(model=gillespy2.Model))
+                Future releases of GillesPy2 may not support this feature.
+                """
+            )
             self = ODECSolver(model, resume=resume)
 
+        if model is not None:
+            log.warning('model = gillespy2.model is deprecated. Future releases '
```

Comment:
when I run the code (in a notebook):
`r1 = model.run()`
I get the red warnings:
`2022-03-28 17:17:51,976 - root - WARNING - model = gillespy2.model is deprecated. Future releases of GillesPy2 may not support this feature.
2022-03-28 17:17:51,976 - root - WARNING - model = gillespy2.model is deprecated. Future releases of GillesPy2 may not support this feature.`
(note the double warning)